src/components/dip-switch/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import Board from '../../board'
+
+import './style.css'
+
+type DipSwitchState = {
+  [key: number]: boolean
+}
+
+type DipSwitchProps = {
+  startIndex: number
+  size: number
+}
+
+export class DipSwitch extends React.Component<DipSwitchProps, DipSwitchState> {
+  constructor(props: DipSwitchProps) {
+    super(props)
+    this.state = this.getState()
+  }
+
+  private getState() {
+    const state: DipSwitchState = {}
+    const endIndex = this.props.startIndex + this.props.size - 1
```

Comment:
should never change on runtime so can it be used as readonly property of DipSwitch?

Reply:
Changed.